### PR TITLE
[Serialization] Always list the bridging header before any imports

### PR DESCRIPTION
**Pull request by jrose-apple**
This mirrors how a bridging header is processed when compiling source files: before any of the imports. This is important for LLDB to recreate the source environment as closely as possible to how the compiler does it; in the test case being added involving a non-modular header file, failure to do so resulted in a deserialization cross-reference crash.

Note that Serialization still sorts imports, which normal resolution of imports in source does not do. So we're still not consistent. But this is less important than handling textual includes (bridging headers) before modular imports.

rdar://problem/40471329

### CODE REVIEW

**Comment by jrose-apple** [comment]
Thanks to Adrian for heroically coming up with a reduced test case!

@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

File: validation-test/Serialization/Inputs/bridging-header-first/AmbivalentProtocol.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+// #if __clang__
+//
+// #ifndef AP
+// #define AP
```
Comment (jrose-apple) [inline validation-test/Serialization/Inputs/bridging-header-first/AmbivalentProtocol.h]: Oops, left some testing code in.

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Review by adrian-prantl — COMMENTED** [review]
Whoa, you fixed it? Awesome!

**Review by DougGregor — APPROVED** [review]

**Pull request merged**